### PR TITLE
Backwards compatibility + test fixes

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

**Comment by clebertsuconic** [comment]
@jbertram  why the revert?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java [hunk 1]
```diff
@@ -1068,6 +1074,31 @@ private void deliverStandardMessage(final MessageReference ref, final Message me
...
+   private void applyPrefixForLegacyConsumer(Message message) {
+      /**
+       * check to see if:
+       * 1) This is a "core" connection
+       * 2) The "core" connection belongs to a JMS client
+       * 3) The JMS client is an "old" client which needs address prefixes
+       *
+       * If 1, 2, & 3 are true then apply the "old" prefix for queues and topics as appropriate.
+       */
+      if (session.getRemotingConnection() instanceof CoreRemotingConnection) {
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java]: Can you please cache this during construction, and only apply this if the cached boolean is set?

This is part of the hot path for delivering messages.. so any gain counts.

Comment (jbertram) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java]: Done. Let me know how the new stuff looks.

**Comment by clebertsuconic** [comment]
@jbertram nice!! thanks a lot for the changes.. much better this way.

**Pull request closed without merging**